### PR TITLE
Add `Default` implementation for `TreiberStack`

### PR DESCRIPTION
**Pull request by torkleyy**

### CODE REVIEW

File: src/sync/treiber_stack.rs [hunk 1]
```diff
@@ -77,6 +77,12 @@ impl<T> TreiberStack<T> {
     }
 }
 
+impl<T> Default for TreiberStack<T> {
+    fn default() -> Self {
+        TreiberStack { head: Atomic::null() }
```
Comment (stjepang) [inline src/sync/treiber_stack.rs]: It would be more idiomatic to leave `fn new` intact and implement `Default` by calling `TreiberStack::new()`.

Comment (torkleyy) [inline src/sync/treiber_stack.rs]: Are you sure? I've seen it more often this way actually.

Comment (torkleyy) [inline src/sync/treiber_stack.rs]: At least clippy recommends it the way it's implemented here: https://github.com/Manishearth/rust-clippy/blob/master/clippy_lints/src/new_without_default.rs#L34

Comment (stjepang) [inline src/sync/treiber_stack.rs]: Hmm, interesting. Well, pretty much every `impl Default for ...` in rust-lang/rust relies on `Self::new()`. But I don't care too much either way. :)

Comment (torkleyy) [inline src/sync/treiber_stack.rs]: Oh okay, didn't really contribute to the Rust compiler / libstd yet. If you still want me to change it, just leave another comment.

Comment (stjepang) [inline src/sync/treiber_stack.rs]: Let's change it.

File: Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "crossbeam"
-version = "0.2.10"
+version = "0.2.11"
```
Comment (jeehoonkang) [inline Cargo.toml]: I think we can bump the version when releasing one.

**Comment by torkleyy** [comment]
Okay, I think this is ready for merge now.

**Pull request merged**